integration/syslog_test.go

```diff
@@ -751,13 +704,68 @@ var _ = Describe("Blackbox", func() {
 				},
 			}
 			blackboxRunner.StartWithConfig(blackboxConfig, 1)
-			logFile.WriteString("hello\n")
-			logFile.Sync()
-			logFile.Close()
+			Write(logFile, "hello\n", true, true)
 
 			Eventually(buffer, "5s").Should(gbytes.Say("hello"))
 
 			blackboxRunner.Stop()
 		})
 	})
+
+	Context("When the server uses bad tls", func() {
+		var address string
+		var buffer *gbytes.Buffer
+		var tlsserver TLSSyslogServer
+		var blackboxRunner *BlackboxRunner
+
+		BeforeEach(func() {
+			address = fmt.Sprintf("127.0.0.1:%d", 9090+GinkgoParallelNode())
+			buffer = gbytes.NewBuffer()
+			tlsserver = TLSSyslogServer{
+				Addr:               address,
+				Buffer:             buffer,
+				CertPrefixOverride: "./fixtures/server-bad",
+			}
+			err := tlsserver.Run()
+			Expect(err).ToNot(HaveOccurred())
+			blackboxRunner = NewBlackboxRunner(blackboxPath)
+		})
+
+		AfterEach(func() {
+			tlsserver.Stop()
+		})
+
+		It("doesn't cause blackbox to fail", func() {
```

Comment:
This test is just to future-proof blackbox against being changed to crash on bad TLS errors, is that it?

Reply:
yup